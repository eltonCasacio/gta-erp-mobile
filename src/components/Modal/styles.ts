import styled from 'styled-components/native';
import Theme from '../../styles/Theme';

export const Wrapper = styled.View``;
export const Text = styled.Text``;
