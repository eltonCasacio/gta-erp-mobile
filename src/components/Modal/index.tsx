import React from 'react';
import * as S from './styles';

import Icon from 'react-native-vector-icons/FontAwesome';

import {TouchableOpacity} from 'react-native';
import Theme from '../../styles/Theme';

export type ModalProps = {
  children?: React.ReactNode;
  callback: Function;
};

const Modal = ({children, callback}: ModalProps) => {
  return (
    <S.Wrapper>
      <S.Content>
        <S.IconWrapper>
          <TouchableOpacity onPress={() => callback()}>
            <Icon name="times" size={30} color={Theme.Colors.grayBold} />
          </TouchableOpacity>
        </S.IconWrapper>
        <S.Text>{children}</S.Text>
      </S.Content>
    </S.Wrapper>
  );
};

export default Modal;
