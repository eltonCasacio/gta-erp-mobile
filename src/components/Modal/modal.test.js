import React from 'react';
import renderer from 'react-test-renderer';

import Modal from '.';

describe('<Modal />', () => {
  it('should render default layout', () => {
    const tree = renderer.create(<Modal />);

    expect(tree).toMatchSnapshot();
  });
});
