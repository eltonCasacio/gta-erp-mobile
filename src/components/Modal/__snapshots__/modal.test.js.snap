// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render default layout 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,0.7)",
        "height": 1329,
        "marginBottom": -5,
        "marginLeft": -5,
        "marginRight": -5,
        "marginTop": -5,
        "paddingBottom": 40,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "position": "absolute",
        "width": 750,
      },
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#fafafa",
          "height": "100%",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "100%",
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "marginBottom": 10,
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#707070",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ÔÄç
          </Text>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#707070",
              "fontSize": 22,
            },
          ]
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
